packages/@aws-cdk/aws-sns-subscriptions/test/subs.test.ts

```diff
@@ -85,7 +85,42 @@ test('url subscription (unresolved url with protocol)', () => {
           "TopicName": "topicName"
         }
       },
-      "MyTopicUnresolvedUrlBA127FB3": {
+      "MyTopicTokenSubscription141DD1BE2": {
+        "Type": "AWS::SNS::Subscription",
+        "Properties": {
+          "Endpoint": "{{resolve:secretsmanager:my-secret:SecretString:::}}",
+          "Protocol": "https",
+          "TopicArn": { "Ref": "MyTopic86869434" },
+        }
+      }
+    }
+  });
+});
+
+test('url subscription (double unresolved url with protocol)', () => {
+  const secret = SecretValue.secretsManager('my-secret');
+  const url = secret.toString();
```

Comment:
Not sure why the original code used `SecretVaue` to generate tokens. Let's switch to use `Token.asString()` instead.

same for other tests.

Reply:
Done, thanks for the feedback